### ISSUE TITLE
Envoy WASM singleton service per envoy instance

### ISSUE THREAD

**Issue by NomadXD** [envoyproxy/envoy#15860]
*Title*: *Envoy WASM singleton service per envoy instance*

*Description*:
I have a requirement where I have to make a HTTP request to an external service and pull some data periodically. I implemented a HTTP filter in Rust and doing the HTTP call from the `onTick` in the `RootContext`. But `RootContext` gets created for each worker thread and for each worker thread , a HTTP occur. In the docs, there is an option for a `Singleton` service where I have to specify `singleton: true`. [https://www.envoyproxy.io/docs/envoy/latest/configuration/other_features/wasm_service#config-wasm-service](https://www.envoyproxy.io/docs/envoy/latest/configuration/other_features/wasm_service#config-wasm-service). However the config example is not very descriptive.

Configured as follows as mentioned in the docs. But envoy rejects the config at start up saying `error initializing configuration '/etc/envoy/envoy.yaml': Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:wasm: Cannot find field.) has unknown fields`. Envoy version used in 1.17.1
```yaml
admin:
  access_log_path: /dev/null
  address:
    socket_address:
      address: 0.0.0.0
      port_value: 9000
wasm:
  config:
    config:
      name: "my_plugin"
      vm_config:
        runtime: "envoy.wasm.runtime.v8"
        code:
          local:
            filename: "/usr/local/bin/singleton_service.wasm"
    singleton: true
static_resources:
  listeners:
  - address:
      socket_address:
        address: 0.0.0.0
        port_value: 9095
```

**Comment by NomadXD** [comment]
Configured as follows as mentioned in the docs. But envoy rejects the config at start up saying `error initializing configuration '/etc/envoy/envoy.yaml': Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:wasm: Cannot find field.) has unknown fields`. Envoy version used in 1.17.1
```yaml
admin:
  access_log_path: /dev/null
  address:
    socket_address:
      address: 0.0.0.0
      port_value: 9000
wasm:
  config:
    config:
      name: "my_plugin_s"
      vm_config:
        runtime: "envoy.wasm.runtime.v8"
        code:
          local:
            filename: "/usr/local/bin/singleton_service.wasm"
    singleton: true
static_resources:
  listeners:
  - address:
      socket_address:
        address: 0.0.0.0
        port_value: 9095
```

**Comment by PiotrSikora** [comment]
That's not the way to configure a bootstrap extension, you need something like this:
```yaml
admin:
  access_log_path: /dev/null
  address:
    socket_address:
      address: 0.0.0.0
      port_value: 9000

bootstrap_extensions:
- name: envoy.bootstrap.wasm
  typed_config:
    '@type': type.googleapis.com/envoy.extensions.wasm.v3.WasmService
    singleton: true
    config:
      vm_config:
        runtime: "envoy.wasm.runtime.v8"
        code:
          local:
            filename: "/usr/local/bin/singleton_service.wasm"

static_resources:
  listeners:
  - address:
      socket_address:
        address: 0.0.0.0
        port_value: 9095
```

**Comment by NomadXD** [comment]
@PiotrSikora Thank you. I think the docs for that relevant section has not been updated yet.

**Comment by PiotrSikora** [comment]
@NomadXD patches welcome!

**Comment by NomadXD** [comment]
@PiotrSikora Sure. Would like to contribute. Any specific procedure that I have to follow before sending a PR for the docs ?

**Comment by PiotrSikora** [comment]
No, simply update `docs/root/configuration/other_features/wasm_service.rst` and make sure that it generates properly (e.g. using `./ci/run_envoy_docker.sh './ci/do_ci.sh docs'`).

If you want more about contributing to Envoy in general, there is [CONTRIBUTING.md](https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md) file that you could read.

**Comment by PiotrSikora** [comment]
Fixed in #15914.
